akka-http-caching/src/main/scala/akka/http/caching/CacheJavaMapping.scala

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+
+package akka.http.caching
+
+import akka.annotation.InternalApi
+import akka.http.caching.javadsl.Cache
+import akka.http.impl.util.JavaMapping
+
+/** INTERNAL API */
+@InternalApi
+private[akka] object CacheJavaMapping {
+
+  def cacheMapping[JK, JV, SK <: JK, SV <: JV] =
+    new JavaMapping[Cache[JK, JV], akka.http.caching.scaladsl.Cache[SK, SV]] {
+      def toScala(javaObject: Cache[JK, JV]): akka.http.caching.scaladsl.Cache[SK, SV] =
+        javaObject.asInstanceOf[akka.http.caching.scaladsl.Cache[SK, SV]]
+
+      def toJava(scalaObject: akka.http.caching.scaladsl.Cache[SK, SV]): Cache[JK, JV] =
+        scalaObject.asInstanceOf[Cache[JK, JV]]
```

Comment:
Due to the casts here I put this note in the caching docs:

> Akka HTTP cache implementations must currently implement both the Java and Scala cache interface. As such, you are currently encouraged to implement new caches in Scala.

Is my understanding correct?

Reply:
Just now noticed the `@DoNotInherit` on the `Cache` API. Hmm.

Reply:
I thought it's the same as the http model, in which we don't don't want to guarantee we won't add methods; and adding methods would be binary incompatible for users extending the cache type, thus this restriction

Reply:
Yes makes sense. I will update the documentation.